packages/transaction-controller/src/utils/layer1-gas-fee-flow.ts

```diff
@@ -77,3 +81,50 @@ async function getTransactionLayer1GasFee({
     return undefined;
   }
 }
+
+/**
+ * Build transactionParams to be used in the unserialized transaction.
+ *
+ * @param transactionMeta - The transaction to build transactionParams.
+ * @returns The transactionParams for the unserialized transaction.
+ */
+function buildTransactionParams(
+  transactionMeta: TransactionMeta,
+): TransactionMeta['txParams'] {
+  return {
+    ...omit(transactionMeta.txParams, 'gas'),
+    gasLimit: transactionMeta.txParams.gas,
+  };
+}
+
+/**
+ * This produces a transaction whose information does not completely match an
+ * Optimism transaction — for instance, DEFAULT_CHAIN is still 'mainnet' and
+ * genesis points to the mainnet genesis, not the Optimism genesis — but
+ * considering that all we want to do is serialize a transaction, this works
+ * fine for our use case.
+ *
+ * @param transactionMeta - The transaction to build an unserialized transaction for.
+ * @returns The unserialized transaction.
+ */
+function buildTransactionCommon(transactionMeta: TransactionMeta) {
+  return Common.custom({
+    chainId: new BN(remove0x(transactionMeta.chainId), 16) as unknown as number,
+    // Optimism only supports type-0 transactions; it does not support any of
+    // the newer EIPs since EIP-155. Source:
+    // <https://github.com/ethereum-optimism/optimism/blob/develop/specs/l2geth/transaction-types.md>
+    defaultHardfork: Hardfork.London,
+  });
+}
+
+/**
+ * Build an unserialized transaction for a transaction.
+ *
+ * @param transactionMeta - The transaction to build an unserialized transaction for.
+ * @returns The unserialized transaction.
+ */
+export function buildUnserializedTransaction(transactionMeta: TransactionMeta) {
```

Comment:
These three functions feel very specific to the Optimism flow as it's the arguments that the Optimism Oracle requires.

Would they be better placed inside the flow itself?

Reply:
Relocated all helpers into flow file https://github.com/MetaMask/core/pull/4055/commits/e912046910ed69c47861ac1121eedc1426c92be6